task_sdk/tests/execution_time/test_supervisor.py

```diff
@@ -762,14 +764,15 @@ def watched_subprocess(self, mocker):
         )
 
     @pytest.mark.parametrize(
-        ["message", "expected_buffer", "client_attr_path", "method_arg", "mock_response"],
+        ["message", "expected_buffer", "client_attr_path", "method_arg", "mock_response", "decoded_buffer"],
```

Comment:
The `decoded_buffer` parameter overrides the comparison in https://github.com/apache/airflow/pull/45344/files#diff-f55b9ba6be46de22b2838085bf1e0aa4193797c3ff7d0cdc4e06a626c96df1baR970-R971. The idea is that when the supervisor a response to the buffer, if `decoded_buffer` is present, it is used for comparison instead of `mock_response`. This is logical because for cases like these when the client responds 404 and we do not pass that all the way to task runner but transform it to something else like in this case (client responded 404, but we sent XcomResponse with value=None).

Reply:
Not applicable, handled differently